test/nodes/test_shaper.py

```diff
@@ -0,0 +1,563 @@
+from haystack import Pipeline, Document
+
+
+def test_basic_function_invocation(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+            - name: shaper
+              params:
+                inputs:
+                    query:
+                      func: expand
+                      output: questions
+                      params:
+                        expand_target: query
+                        size:
```

Comment:
Just my thoughts from looking at the yaml definition. I would be confused as a user when I would need to provide a `params` definition for the chosen function (in this case `func: expand`). 

I see why we need to specify the `size` although I don't understand why we need to specify the `expand_target: query` since this `func` is already under the `query:` input. 

In general though I think we would need docs showing these examples on the website and docs on the available `func`'s and their parameters so users would know how to specify them in a yaml file.

Reply:
Yeah, good point. We handle the use case where no param is specified, and that works well. See unit tests. But I think we want to make this connection explicit.